README.md

```diff
@@ -42,3 +42,10 @@ To install in a remote cluster, it is assumed that an argocd instance is already
 2. `kubectl create ns argocd && kubectl apply -f manifests/argocd-install/app-of-apps-application.yaml`
 
 3. Coffee time. It should all be green afte some minutes.
+
+If you have installed on OpenShift, at this time you also need to manually create a token for Grafana to access the thanos-query instance. The token should be stored in a secret that is referenced by the Grafana datasource:
+
+```bash
+TOKEN="$(oc -n monitoring create token thanos-query --duration=31536000s)"
+oc -n monitoring create secret generic thanos-query-token --from-literal=token=$TOKEN
+```
```

Comment:
There's a way to generate a long lived token declaratively. If a Secret is created with the annotation `kubernetes.io/service-account.name=<grafana-serviceaccount-name>`, a token with no expiration is generated automatically within such Secret.

Example:

```
apiVersion: v1
kind: Secret
metadata:
  name: thanos-datasource-auth
  annotations:
    # Requires grafana instance to exist
    argocd.argoproj.io/sync-wave: "grafana-instance-wave+1"
    # this will generate a long-lived token for the service account in the annotation
    kubernetes.io/service-account.name: grafana-sa
type: kubernetes.io/service-account-token
```

Even though k8s documentation recommends using short-lived tokens, a long-lived one is probably justified in this case. When monitoring stops working after a year, chances are you won't remember that this token needs to be recreated.

Docs to long-lived tokens feature: https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#create-token